### PR TITLE
Refactor to store projects in BKProject format

### PR DESCRIPTION
**Pull request by JCFarrow**
Bible Karaoke now imports all projects as BKProjects on startup.

### CODE REVIEW

File: main/sources/hear-this.ts [hunk 1]
```diff
@@ -29,46 +42,146 @@ class HearThis implements ProjectSource {
...
+    }
     return book;
   }
 
-  makeChapter(projectName: string, bookName: string, name: string, directory: string): Chapter {
-    const chapter = new Chapter();
-    chapter.name = parseInt(name).toString();
-    const chapterFiles = fs.readdirSync(path.join(directory, projectName, bookName, name));
-    chapter.audioFiles = chapterFiles
-      .filter((file: string) => isValidAudioFile(file, DEFAULT_HEARTHIS_XML_FILE, fileFilters.audio[0].extensions))
```
Comment (irahopkinson) [inline main/sources/hear-this.ts:63]: You have removed the use of the file filters. Do you know for certain that HearThis files will only be WAV and not ever MP3?

File: main/sources/hear-this.ts [hunk 2]
```diff
@@ -29,46 +42,146 @@ class HearThis implements ProjectSource {
...
+      name: chapterName,
+      audio: {
+        files: [],
+        length: 0,
+      },
+      segments: [],
+    };
+    const sourceChapterDir = path.join(directory, projectName, bookName, chapterName);
+    const chapterFiles = fs.readdirSync(sourceChapterDir, 'utf8');
+    const audioFiles = chapterFiles.filter((file: string) => file.endsWith('.wav'));
```
Comment (irahopkinson) [inline main/sources/hear-this.ts:110]: What if the extension is in caps? Perhaps use `file.toLowerCase().endsWith('.wav')`.

File: main/electron.ts [hunk 3]
```diff
@@ -135,8 +123,12 @@ export function handleSubmission(): void {
     let response: string | Error;
     try {
       // ToDo: move this to the frontend and pass a subset of the selected chapters across the IPC.
```
Comment (irahopkinson) [inline main/electron.ts:125]: This comment should be removed now that you have done it! Yeh!

File: src/App/store/AppState.ts [hunk 4]
```diff
@@ -26,20 +26,6 @@ const SAMPLE_VERSES = [
     'And there was evening and there was morning, the first day.',
 ];
 
-const list = <T = Project>(dict: { [name: string]: T }, sortKey = 'name'): T[] => _.sortBy(_.values(dict), sortKey);
-
-const dict = <T = Project | Book | Chapter>(
```
Comment (irahopkinson) [inline src/App/store/AppState.ts:31]: So pleased you could get rid of `dict` and `list`. Well done.

File: main/import/hearThis/test/scenarios/index.ts [hunk 5]
```diff
@@ -1,5 +0,0 @@
-import { scenario1 } from './scenario1';
-import { scenario2 } from './scenario2';
-import { scenario3 } from './scenario3';
-
-export const scenarios = [scenario1, scenario2, scenario3];
```
Comment (irahopkinson) [inline main/import/hearThis/test/scenarios/index.ts:5]: Are all these _import_ scenarios now covered by _source_ test scenarios?

**Review by irahopkinson — CHANGES_REQUESTED** [review]
I'm very excited to be at this point, Thanks for all this work. I have a few questions.

File: main/sources/hear-this.ts (hunk 1 above)
Comment (JCFarrow) [reply main/sources/hear-this.ts:63]: I figured that HearThis would always generate audio with the same type of extension (lowercase .wav). But it's possible that a future update could change this so I have re-added the `checkValidAudioFile` function.

File: main/sources/hear-this.ts (hunk 2 above)
Comment (JCFarrow) [reply main/sources/hear-this.ts]: Done

File: main/electron.ts (hunk 3 above)
Comment (JCFarrow) [reply main/electron.ts]: Done. I wasn't sure if this was what I had done or if it was referring to something else.

File: main/import/hearThis/test/scenarios/index.ts (hunk 5 above)
Comment (JCFarrow) [reply main/import/hearThis/test/scenarios/index.ts:5]: These three scenarios are testing going from a `convertProject` to a `BKProject` which no longer happens. And they use the same test data that `fullImport.integration` uses (`main/test`) so just having the full import test should cover them.

**Review by irahopkinson — APPROVED** [review]
👍 LGTM, well done.

**Pull request merged**